src/pybind/mgr/dashboard/frontend/src/app/shared/components/copy2clipboard-button/copy2clipboard-button.component.ts

```diff
@@ -1,31 +1,33 @@
-import { Directive, ElementRef, HostListener, Input, OnInit, Renderer2 } from '@angular/core';
+import { Component, HostListener, Input } from '@angular/core';
 
 import { detect } from 'detect-browser';
 import { ToastrService } from 'ngx-toastr';
 
-@Directive({
-  selector: '[cdCopy2ClipboardButton]'
+@Component({
+  selector: 'cdCopy2ClipboardButton',
+  templateUrl: './copy2clipboard-button.component.html',
+  styleUrls: ['./copy2clipboard-button.component.scss']
 })
-export class Copy2ClipboardButtonDirective implements OnInit {
+export class Copy2ClipboardButtonComponent {
   @Input()
-  private cdCopy2ClipboardButton: string;
+  private source: string;
 
   constructor(
-    private elementRef: ElementRef,
-    private renderer: Renderer2,
+    // private elementRef: ElementRef,
+    // private renderer: Renderer2,
     private toastr: ToastrService
   ) {}
 
-  ngOnInit() {
-    const iElement = this.renderer.createElement('i');
-    this.renderer.addClass(iElement, 'fa');
-    this.renderer.addClass(iElement, 'fa-clipboard');
-    this.renderer.setAttribute(iElement, 'title', $localize`Copy to clipboard`);
-    this.renderer.appendChild(this.elementRef.nativeElement, iElement);
-  }
+  // ngOnInit() {
```

Comment:
Are these comments needed?

Reply:
Ahh, no longer; I'll delete them on pushing the changes to the spec.